src/SimpleGraphs/generators/randgraphs.jl

```diff
@@ -1422,3 +1422,53 @@ function random_orientation_dag(
     end
     return g2
 end
+
+"""
+    bernoulli_graph(Λ)
+
+Given the symmetric matrix ``\\Lambda \\in [0,1]^{n \\times n}``, return a Bernoulli graph with ``n`` vertices. Each edge ``(i,j)`` exists with probability ``\\operatorname{Bernoulli}(\\Lambda[i,j])``.
+"""
+function bernoulli_graph(
+    Λ::Matrix{Float64};
+    rng::Union{Nothing,AbstractRNG}=nothing,
+    seed::Union{Nothing,Integer}=nothing,
+)
+    @assert size(Λ)[1] == size(Λ)[2] "The probability matrix must be a square matrix!"
+    n = size(Λ)[1]
+    rng = rng_from_rng_or_seed(rng, seed)
+    A = SimpleGraph(n)
+    for j in 1:n
+        for i in (j + 1):n
+            if Bool(randbn(1, Λ[i, j]; rng=rng))
+                add_edge!(A, i, j)
+            end
+        end
+    end
+    return A
+end
+
+"""
+rho_correlated_bernoulli_graphs(Λ,ρ)
+
+Given the symmetric matrix ``\\Lambda \\in [0,1]^{n \\times n}`` and a real number ``\\rho \\in [0,1]`` return two ``\\rho``-correlated Bernoulli graphs with ``n`` vertices.
+"""
+function rho_correlated_bernoulli_graphs(
+    Λ::Matrix{Float64},
+    ρ::Float64;
+    rng::Union{Nothing,AbstractRNG}=nothing,
+    seed::Union{Nothing,Integer}=nothing,
+)
+    n = size(Λ)[1]
+    rng = rng_from_rng_or_seed(rng, seed)
+    B = SimpleGraph(n)
+    A = bernoulli_graph(Λ; rng=rng, seed=seed)
+    A_adj = Int.(adjacency_matrix(A))
+    for j in 1:n
+        for i in (j + 1):n
+            if Bool(randbn(1, (1 - ρ) * Λ[i, j] + ρ * A_adj[i, j]; rng=rng))
+                add_edge!(B, i, j)
+            end
+        end
+    end
+    return A, B
```

Comment:
We should probably document somewhere that what is returned is a tuple of two graphs